src/components/pages/devices/flyouts/deviceDelete/deviceDelete.js

```diff
@@ -0,0 +1,141 @@
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) this.subscription.unsubscribe();
+  }
+
+  populateDevicesState = (devices = []) => {
+    const physDevices = devices.filter(({ isSimulated }) => !isSimulated);
+    this.setState({ physicalDevices: physDevices, containsSimulatedDevices: (physDevices.length !== devices.length) });
+  }
+
+  deleteDevices = () => {
+    this.setState({ isPending: true });
+
+    this.subscription = Observable.from(this.state.physicalDevices)
+      .flatMap(({ id }) =>
+        IoTHubManagerService.deleteDevice(id)
+          .map(() => id)
+      )
+      .subscribe(
+        deletedDeviceId  => {
+          this.setState({ successCount: this.state.successCount + 1 });
+          this.props.deleteDevice(deletedDeviceId);
+        },
+        error => this.setState({ error, isPending: false, changesApplied: true }),
+        () => this.setState({ isPending: false, changesApplied: true })
+      );
+
+  }
+
+  render() {
+    const { t, onClose } = this.props;
+    const {
+      physicalDevices,
+      containsSimulatedDevices,
+      isPending,
+      error,
+      successCount,
+      changesApplied
+    } = this.state;
+
+    const summaryCount = changesApplied ? successCount : physicalDevices.length;
+    const summaryMessage = changesApplied ? t('devices.delete.applySuccess') : t('devices.delete.affected');
+
+    return (
+      <Flyout>
+        <FlyoutHeader>
+          <FlyoutTitle>{t('devices.delete.title')}</FlyoutTitle>
+          <FlyoutCloseBtn onClick={onClose} />
+        </FlyoutHeader>
+        <FlyoutContent>
+          <div className="device-delete-container">
+            <div className="device-delete-header">{t('devices.delete.header')}</div>
+            <div className="device-delete-descr">{t('devices.delete.description')}</div>
+            {
+              containsSimulatedDevices &&
+              <div className="simulated-device-selected">
+                <Svg path={svgs.infoBubble} className="info-icon" />
+                {t('devices.delete.simulatedNotSupported')}
+              </div>
+            }
+            <div className="device-delete-summary-container">
+              <div>
+                Summary
+              </div>
+              <div className="device-delete-affected">
+                <div className="device-delete-count">{summaryCount}</div>
+                {summaryMessage}
+              </div>
+            </div>
+            {
+              error &&
+              <div className="device-delete-error">
+                <ErrorMsg>{error}</ErrorMsg>
+              </div>
+            }
+            {
+              !changesApplied &&
+              <BtnToolbar className="tools-preApply">
+                {this.state.isPending && <Indicator />}
+                <Btn svg={svgs.trash} primary={true} disabled={isPending || physicalDevices.length === 0} onClick={this.deleteDevices}>{t('devices.delete.apply')}</Btn>
```

Comment:
I would do `primary={true}` for the buttons. It should either be a single attribute with an implicit true value or it should be something like `btnType="primary"`

Reply:
I thought this was how you told me to do it to begin with.  Otherwise, tell me this 4 days ago.

Reply:
I'm not sure I agree with an essentially free form string value like btnType would be.  Its really no better than using className and joining them under the covers.